kornia/contrib/sam/predictor.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+from typing import Any
+
+import torch
+
+from kornia.augmentation import AugmentationSequential, LongestMaxSize
+from kornia.core import Device, Tensor
+from kornia.core.check import KORNIA_CHECK_SHAPE
+from kornia.geometry.boxes import Boxes
+from kornia.geometry.keypoints import Keypoints
+
+from .architecture import Sam
+from .model import SamPrediction
+
+
+class SamPredictor:
+    """Uses SAM to calculate the image embedding for an image, and then allow repeated, efficient mask prediction
+    given prompts.
+
+    Args:
+        sam_model: The model to use for mask prediction.
+    """
+
+    def __init__(self, sam_model: Sam) -> None:
+        super().__init__()
+        self.model = sam_model
+        self._longside_size = self.model.image_encoder.img_size
+        transforms = (LongestMaxSize(self._longside_size),)
+        self.tfs = AugmentationSequential(*transforms, same_on_batch=True)
+
+    def device(self) -> Device:
+        return self.model.device
+
+    @torch.no_grad()
+    def __call__(
```

Comment:
initially, I hadn't seen the need, but maybe this can be very useful for the auto mask API

also, seems to be more simple to just call with the input image than using "set -> predict -> reset" pipeline

Reply:
I think we should have both modes. With this model I see the main use case of I give one image and I want to make multiple queries in separated calls. E.g. to integrate into a labelling service.